Bottom: 3e90b33543a7ad7f7e207d2b8560406cce1239f4
Top:    938192a2b549d59bac7606a084f4e9a17a6ab9b4
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-23 16:56:13 +0100

draw.c: Change DrawTop() to DrawPPV() taking a side parameter

(PPV stands for Pins, Pads and Vias)


---

diff --git a/src/draw.c b/src/draw.c
index 7380beb..b0bd9f6 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -83,7 +83,7 @@ static const BoxType *clip_box = NULL;
  */
 static void Redraw (bool, BoxTypePtr);
 static void DrawEverything (BoxTypePtr);
-static void DrawTop (const BoxType *);
+static void DrawOuterPPV (int side, const BoxType *);
 static int DrawLayerGroup (int, const BoxType *);
 static void DrawPinOrViaLowLevel (PinTypePtr, bool);
 static void DrawPlainPin (PinTypePtr, bool);
@@ -312,14 +312,15 @@ static void
 PrintAssembly (const BoxType * drawn_area, int side_group, int swap_ident)
 {
   int save_swap = SWAP_IDENT;
+  int side = swap_ident ? SOLDER_LAYER : COMPONENT_LAYER;
 
   gui->set_draw_faded (Output.fgGC, 1);
-  SWAP_IDENT = swap_ident;
   DrawLayerGroup (side_group, drawn_area);
-  DrawTop (drawn_area);
+  DrawOuterPPV (side, drawn_area);
   gui->set_draw_faded (Output.fgGC, 0);
 
   /* draw package */
+  SWAP_IDENT = swap_ident;
   DrawSilk (swap_ident,
 	    swap_ident ? solder_silk_layer : component_silk_layer,
 	    drawn_area);
@@ -384,13 +385,12 @@ DrawEverything (BoxTypePtr drawn_area)
 	  if (DrawLayerGroup (group, drawn_area) && !gui->gui)
 	    {
 	      r_search (PCB->Data->pin_tree, drawn_area, NULL, pin_callback, NULL);
-	      r_search (PCB->Data->via_tree, drawn_area, NULL, via_callback, NULL);
-	      /* draw element pads */
 	      if (group == component || group == solder)
 		{
                   side = (group == solder) ? SOLDER_LAYER : COMPONENT_LAYER;
 		  r_search (PCB->Data->pad_tree, drawn_area, NULL, pad_callback, &side);
 		}
+	      r_search (PCB->Data->via_tree, drawn_area, NULL, via_callback, NULL);
 
 	      /* draw holes */
 	      plated = -1;
@@ -402,9 +402,11 @@ DrawEverything (BoxTypePtr drawn_area)
   if (TEST_FLAG (CHECKPLANESFLAG, PCB) && gui->gui)
     return;
 
+  side = SWAP_IDENT ? SOLDER_LAYER : COMPONENT_LAYER;
+
   /* Draw pins, pads, vias below silk */
   if (gui->gui)
-    DrawTop (drawn_area);
+    DrawOuterPPV (side, drawn_area);
   else
     {
       HoleCountStruct hcs;
@@ -566,13 +568,11 @@ pad_callback (const BoxType * b, void *cl)
 }
 
 /* ---------------------------------------------------------------------------
- * draws pins pads and vias
+ * draws outer layer pins pads and vias
  */
 static void
-DrawTop (const BoxType * screen)
+DrawPPV (int side, const BoxType * screen)
 {
-  int side = SWAP_IDENT ? SOLDER_LAYER : COMPONENT_LAYER;
-
   if (PCB->PinOn || doing_assy)
     {
       /* draw element pins */
